servers/zms/src/main/java/com/yahoo/athenz/zms/notification/NotificationManager.java

```diff
@@ -0,0 +1,162 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.athenz.zms.notification;
+
+import com.yahoo.athenz.common.server.notification.Notification;
+import com.yahoo.athenz.common.server.notification.NotificationService;
+import com.yahoo.athenz.common.server.notification.NotificationServiceFactory;
+import com.yahoo.athenz.zms.*;
+import com.yahoo.athenz.zms.store.AthenzDomain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class NotificationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationManager.class);
+
+    NotificationServiceFactory notificationServiceFactory;
+    NotificationService notificationService;
+    ScheduledExecutorService scheduledExecutor;
+    private DBService dbService;
+
+    public NotificationManager(DBService dbService) {
+
+        this.dbService = dbService;
+        String notificationServiceFactoryClass = System.getProperty(ZMSConsts.ZMS_PROP_NOTIFICATION_SERVICE_FACTORY_CLASS);
+        NotificationServiceFactory notificationServiceFactory;
+        try {
+            notificationServiceFactory = (NotificationServiceFactory) Class.forName(notificationServiceFactoryClass).newInstance();
+            notificationService = notificationServiceFactory.create();
+
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            LOGGER.error("Invalid NotificationServiceFactory class: " + notificationServiceFactoryClass + " error: " + e.getMessage());
+            throw new IllegalArgumentException("Invalid notification service factory");
+        }
+
+        init();
+    }
+
+    private void init() {
+        if (notificationService != null) {
+            scheduledExecutor = Executors.newScheduledThreadPool(1);
+            scheduledExecutor.scheduleAtFixedRate(new PendingMembershipApprovalReminder(), 0, 1, TimeUnit.DAYS);
+        }
+    }
+
+    public NotificationManager(DBService dbService, NotificationServiceFactory notificationServiceFactory) {
+
+        this.dbService = dbService;
+        this.notificationServiceFactory = notificationServiceFactory;
+        notificationService = this.notificationServiceFactory.create();
+
+        init();
+    }
+
+    void shutdown() {
+        if (scheduledExecutor != null) {
+            scheduledExecutor.shutdownNow();
+        }
+    }
+
+    public void generateAndSendPostPutMembershipNotification(String domain, String org, Boolean auditEnabled, Boolean selfserve, Map<String, String> details) {
+
+        Set<String> recipients = new HashSet<>();
```

Comment:
it's possible that we don't have any notification service configured. So before doing anything else we must check if the service is null or not and it's null, return without doing any work.